### DISCUSSION TITLE
custom vim bindings (<leader>)

### DISCUSSION THREAD

**Discussion by Josiah-tan** [opensteno/plover#1467] [Ideas]
So I think that normal vim bindings (diw, ysip", 14j) are handled pretty well in the discussion from [vim navigation](https://github.com/openstenoproject/plover/discussions/1350).

But I want some sort of system for custom bindings that is easy to remember, so I am wondering what people have come up with before I decide to commit to anything.

- For example it be nice to have an easy way to stroke all of these in preferably 1 key stroke:
  - note: that it's fine to do it in 2, maybe as a preparation stroke, I don't mind it too much
  - (of course this can easily be made to work in any vim mode as well which was discussed [here](https://github.com/openstenoproject/plover/discussions/1350#discussioncomment-1905781))

```vim
nnoremap <leader>sg <cmd>lua require('telescope.builtin').grep_string()<cr>
nnoremap <leader>ht <cmd>lua require('telescope.builtin').help_tags()<cr>
nnoremap <leader>gd <cmd>lua require('telescope.builtin').lsp_definitions()<cr>
" (yes I'm aware that lua mappings are supported in nvim 0.7 now, don't judge to much plz)
```

- I also generally try to keep my keymaps less than 3 characters long (including the leader key)

- it would be nice if I don't have to memorize new briefs for every single new key mapping
  - like STKPWR-FGT for grep_string for example

# some ideas (not in any particular order):
### Method 1
- using command + letter from [plover-vim](https://github.com/Josiah-tan/plover-vim)
  - first stroke outputs: {#escape space someLetter} (from plover-vim)
    - (not currently in the defaults but easily extendable)
  - second stroke outputs another letter (can be just finger spelt)
 
### Method 2
- unique stroke + double sided finger spelling (similar to user2's [easy motion](https://github.com/openstenoproject/plover/discussions/1350#discussion-3408223) code)
  - first stroke outputs nothing / takes you to normal mode
  - second stroke spells two letters
- compared to method 3, the finger spelling does not have to be unique on the right hand, missing letters can be created more easily

### Method 3

- finger spell on both sides at the same time
  - left hand, same as normal
    - but with SK for E, WR for U, SKW for I (to free up more of the right hand)
  - right hand, 26 unique chords for every letter of the alphabet?

**Comment by user202729** [comment]
I don't have a very-good system for these ad hoc things, just STPR + random nonconflicting (*) combination on right hand, although I try to group them somewhat logically.

For now I think I have at most 20 such ad hoc (frequently used) strokes? Definitely doesn't need the space of 2 strokes.

As a result, sometimes I know the *steno stroke* for an action but not the *normal keyboard key chain*.

Possible improvement, if the "normal keyboard" part is made *later* arbitrarily, it can be made that somewhat resemble the stroke. (only matters if sometimes you want to quickly stroke something without a steno keyboard around)

(*) may cause conflict later if the "systematic" part is extended.
